src/main/java/net/minecraftforge/common/extensions/IForgeEnchantment.java

```diff
@@ -28,4 +28,19 @@ default float getDamageBonus(int level, MobType mobType, ItemStack enchantedItem
     {
         return self().getDamageBonus(level, mobType);
     }
+
+    /**
+     * Should the book for this enchantment appear in the creative tab
+     * @param book Items.ENCHANTED_BOOK normally, but modded items may also use this hook.
+     * @param tab The creative tab that items are being added to
+     * @return true to put the item variant in the tab.
+     */
+    default boolean allowedInCreativeTab(net.minecraft.world.item.Item book, net.minecraft.world.item.CreativeModeTab tab) {
+        if(!self().isAllowedOnBooks())
+            return false;
+        else if(tab == net.minecraft.world.item.CreativeModeTab.TAB_SEARCH)
+            return self().category != null;
+        else
+            return tab.hasEnchantmentCategory(self().category);
```

Comment:
Please do not use if statements without braces.
Please ensure the correct formatting (space missing after `if`)